tensorflow_lite_support/cc/test/task/processor/image_preprocessor_test.cc

```diff
@@ -0,0 +1,147 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow_lite_support/cc/task/processor/image_preprocessor.h"
+
+#include <fstream>
+#include <memory>
+
+#include "absl/flags/flag.h"
+#include "absl/status/status.h"
+#include "absl/strings/cord.h"
+#include "tensorflow/lite/core/shims/cc/shims_test_util.h"
+#include "tensorflow/lite/kernels/builtin_op_kernels.h"
+#include "tensorflow/lite/mutable_op_resolver.h"
+#include "tensorflow_lite_support/cc/port/gmock.h"
+#include "tensorflow_lite_support/cc/port/gtest.h"
+#include "tensorflow_lite_support/cc/port/status_matchers.h"
+#include "tensorflow_lite_support/cc/task/core/task_api_factory.h"
+#include "tensorflow_lite_support/cc/task/core/task_utils.h"
+#include "tensorflow_lite_support/cc/task/core/tflite_engine.h"
+#include "tensorflow_lite_support/cc/task/vision/utils/frame_buffer_common_utils.h"
+#include "tensorflow_lite_support/cc/test/test_utils.h"
+#include "tensorflow_lite_support/examples/task/vision/desktop/utils/image_utils.h"
+
+namespace tflite {
+namespace task {
+namespace processor {
+namespace {
+
+using ::testing::ElementsAreArray;
+using ::testing::HasSubstr;
+using ::testing::Optional;
+using ::tflite::support::kTfLiteSupportPayload;
+using ::tflite::support::StatusOr;
+using ::tflite::support::TfLiteSupportStatus;
+using ::tflite::task::JoinPath;
+using ::tflite::task::core::PopulateTensor;
+using ::tflite::task::core::TaskAPIFactory;
+using ::tflite::task::core::TfLiteEngine;
+using ::tflite::task::vision::DecodeImageFromFile;
+using ::tflite::task::vision::FrameBuffer;
+using ::tflite::task::vision::ImageData;
+
+constexpr char kTestDataDirectory[] =
+    "tensorflow_lite_support/cc/test/testdata/task/vision/";
+
+constexpr char kDilatedConvolutionModelWithMetaData[] = "dilated_conv.tflite";
```

Comment:
Does this model has metadata? Otherwise remove `WithMetaData` from the name.

Reply:
Yep, it has metadata. Since the model is `kTfLiteFloat32` type, metadata is mandatory. I used this [script](https://gist.github.com/jonpsy/500c1448fbf7434feabb5b9b804ee331)  to populate it. I've also put the issue number in the metadata.

Reply:
I see. You can set 
```
input_normalization.options.mean = [0]
input_normalization.options.std = [1]
```
to by-pass normalization. Then comparing with the golden image will be much easier.